app/code/core/Mage/Adminhtml/controllers/GridController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2023 The OpenMage Contributors (https://www.openmage.org)
+ * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Backend grid controller
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ */
+class Mage_Adminhtml_GridController extends Mage_Adminhtml_Controller_Action
+{
+    /**
+     * ACL resource
+     * @see Mage_Adminhtml_Controller_Action::_isAllowed()
+     */
+    public const ADMIN_RESOURCE = 'catalog/products';
+
+    /**
+     * Save grid columns order action
+     */
+    public function saveColumnOrderAction()
+    {
+        $gridId = $this->getRequest()->getPost('gridId');
+        $data = $this->getRequest()->getPost('data');
+        $reset = $this->getRequest()->getPost('reset');
+        if (!$gridId || (!$data && !$reset)) {
+            return false;
+        }
+
+        /** @var Mage_Adminhtml_Helper_Widget_Grid_Config $_advancedGridHelper  */
+        $_advancedGridHelper = Mage::helper('adminhtml/widget_grid_config');
+        $_advancedGridHelper
+            ->setGridId($gridId)
+            ->saveOrderColumns($data);
+    }
+
+    /**
+     * Check is allowed access to action
+     *
+     * @return bool
+     */
+    protected function _isAllowed()
+    {
+        return Mage::getSingleton('admin/session')->isAllowed(static::ADMIN_RESOURCE);
```

Comment:
@empiricompany I've introduced this mega basic ACL check, it checks for "manage products". I've been thinking about it for a while. The problem is, do we want to introduce an ACL for every grid? It seems very overkill.

At the moment there has never been an ACL for modifying the backend interface, since it was never possible before.

Reply:
I was doing some tests today and I was thinking, what if...
- system->configuration page has a button "enable column rearrangement"
- the buttons just sets a variable in the session
- the current logged-in user can do the d&d on the grids

with this we wouldn't need ACL at all and it would remove the 2 buttons from the grid page.

what do you think?

Reply:
I'm not a fun of storing this in session I want to keep same behavior across browser with the same account

I prefer to combine your proposed system config select to enable rearrangement of columns across all grids and set this to No by default


then add an acl per user role to permit to change this config and others

and also check it in grid controller action

in my next time slot dedicated to this project I propose you this behavior